src/wxflow/fsutils.py

```diff
@@ -45,22 +56,35 @@ def chdir(path):
         do_thing_2
     """
     cwd = os.getcwd()
+    # Try to change paths.
     try:
         os.chdir(path)
+    except OSError:
+        raise OSError(f"Failed to change directory to ({path})")
+
+    # If successful, yield to the calling "with" statement.
+    try:
```

Comment:
would this always `yield`? the finally is never executed

Reply:
Yes, this will always `yield`.  The `finally` section is executed even if something in the `with` clause raises an error.  I added a test to confirm this.